conf/spark-env.sh

```diff
@@ -72,6 +72,16 @@ else
     echo "spark-env: No SPARK_MESOS_KRB5_CONF_BASE64 decoded" >&2
 fi
 
+echo ${PWD}
+
+if [ -n "${STATSD_UDP_HOST}" ] && [ -n "${STATSD_UDP_PORT}" ]; then
+    sed -e "s/<STATSD_UDP_HOST>/${STATSD_UDP_HOST}/g" \
+        -e "s/<STATSD_UDP_PORT>/${STATSD_UDP_PORT}/g" \
+        ${PWD}/conf/metrics.properties.template >${PWD}/conf/metrics.properties
```

Comment:
Hi @hectorj2f , Sam ran an image with this PR and encountered an error:
> /opt/spark/dist/conf/spark-env.sh: line 78: /mnt/mesos/sandbox/conf/metrics.properties: No such file or directory
It looks like $PWD = "/mnt/mesos/sandbox", whereas the config file should be in "/opt/spark/dist". What do you think? The strange thing is that the docker image that you provided me didn't have this error (mesosphere/spark-dev:931ca56273af913d103718376e2fbc04be7cbde0).

Reply:
BTW, the docker image from CI is tagged with the commit SHA. So, the tag should be "5a1250c1b3c21f6c5c0553fd4d6d407beb5fcb39" (your last commit), not the one above.

Reply:
@susanxhuynh To generate the docker image, I simply typed `docker-dist` from your Makefile. If the image doesn't have a proper tag, then there is a bug in the Makefile

Reply:
> It looks like $PWD = "/mnt/mesos/sandbox", whereas the config file should be in "/opt/spark/dist". What do you think? The strange thing is that the docker image that you provided me didn't have this error (mesosphere/spark-dev:931ca56273af913d103718376e2fbc04be7cbde0).

I'd like to know how he generates the image 🤔 , I tried some of your old images and all have the `PWD=/opt/spark/dist`.

Reply:
@hectorj2f OK, I think I know what happened. If you look at https://github.com/mesosphere/spark-build/blob/5a1250c1b3c21f6c5c0553fd4d6d407beb5fcb39/conf/spark-env.sh#L15, there's a line that is `cd $MESOS_SANDBOX`. If you are inside a Mesos task (like running a container in Marathon), MESOS_SANDBOX=/mnt/mesos/sandbox. If you are not in a Mesos task, MESOS_SANDBOX is not defined, so you will stay in the default directory of the container (/opt/spark/dist). The Spark task is a Mesos task, so it is defined.

You can test it like this: 
- go to the CI test below, click on "Details". That will take you to the TC job: https://teamcity.mesosphere.io/viewLog.html?buildId=1049548&buildTypeId=DcosIo_DcosCommons_Spark_SparkBuildPr
- In Step 3, it does a "make docker dist". The result is:

> [13:33:59][Step 3/8] Successfully tagged mesosphere/spark-dev:5a1250c1b3c21f6c5c0553fd4d6d407beb5fcb39

Then go to Marathon and run that container ^^, with 
- container image=`mesosphere/spark-dev:5a1250c1b3c21f6c5c0553fd4d6d407beb5fcb39`
- command=`conf/spark-env.sh`
- container runtime="UCR".
Then you will see the error in stderr:

> conf/spark-env.sh: line 78: /mnt/mesos/sandbox/conf/metrics.properties: No such file or directory
<img width="1437" alt="screen shot 2018-04-28 at 8 03 49 am" src="https://user-images.githubusercontent.com/22622418/39397855-bd673bf6-4aba-11e8-86ee-ce780ef58ed1.png">

Reply:
I'll apply these changes. What about this comment from slack https://mesosphere.slack.com/archives/CABQ906UR/p1524867727000267 ? What do we need to add to the docker image ?